In [7]:
import pandas as pd

In [8]:
url = "https://raw.githubusercontent.com/pandas-dev/pandas/main/doc/data/titanic.csv"
df = pd.read_csv(url)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [9]:
df.shape


(891, 12)

############################################################################################################################################################################################

<h1 style="text-align:center;"> TITANIC SURVIVAL ANALYSIS </h1>

<h2 style="text-align:center;">(Using Python Pandas and Jupyter Notebook)</h2>

\############################################################################################################################################################################################

### INTRODUCTION:

This project aims to analyze and interpret the Titanic passenger dataset to understand what are the different factors that may have affected survival during the tragic 1912 Titanic disaster. The RMS Titanic was a British passenger liner deemed "practically unsinkable," which sank on its maiden voyage after striking an iceberg on April 14, 1912. I will be using Python Pandas for studying and identifying different patterns which are related to the gender, age, family size, passenger class and fare. I will be using data analysis and visualizations techniques to identify which passengers characteristics were most strongly connected to survival. 

############################################################################################################################################################################################

### DATASET DESCRIPTION:


The dataset used for the project is the Survival dataset from the tragic sinking of RMS Titanic in 1912 which contains about 891 Titanic passengers and 12 columns. The important columns included are Survived, Pclass, Name, Sex, Age, SibSp, Parch, Fare, Cabin and Embarked. The passengers that did not survive use the 0 in the column, whereas the passengers who did survive represents 1.  

############################################################################################################################################################################################

### QUESTIONS:

1. What percentage of passengers survived?
2. Did gender affect survival?
3. Did passenger class affect survival?
4. Did age affect survival?
5. Did fare price affect survival?
6. Did family size affect survival?
7. Which embarkation port had the highest survival rate?
8. What are the main insights from the analysis?

############################################################################################################################################################################################

### DATA CLEANING AND PREPROCESSING

After I had loaded the dataset, I checked the dataset for any missing values or inconsistencies that could modify or make the data visualization. After checking, I identified that Age, Cabin, and Embarked columns contains the most missing values. 
For preparing the cleaned and preprocessed dataset for analysis:

- The missing values from Age were filled using the median
- The missing values in Embarked were filled using the most frequent value
- The cabin column was removed completely due to a large number of missing values. 

After this, I can confirm the dataset is clean and suitable, ready fro analysis. 

In [10]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

############################################################################################################################################################################################